lib/smart_app_launch/smart_stu2_suite.rb

```diff
@@ -210,14 +211,46 @@ class SMARTSTU2Suite < Inferno::TestSuite
       title 'Token Introspection'
       id :smart_token_introspection
-
-      input_instructions <<~INSTRUCTIONS
-        TODO: Instructions for token introspection go here!
-      INSTRUCTIONS
+      description %(
+        # Background
+
+        OAuth 2.0 Token introspection, as described in [RFC-7662](https://datatracker.ietf.org/doc/html/rfc7662), allows
+        an authorized resource server to query an OAuth 2.0 authorization server for metadata on a token.  The
+        [SMART App Launch STU 2.1 Implementation Guide Section on Token Introspection](https://hl7.org/fhir/smart-app-launch/token-introspection.html)
+        states that "SMART on FHIR EHRs SHOULD support token introspection, which allows a broader ecosystem of resource servers
+        to leverage authorization decisions managed by a single authorization server."
+  
+        # Test Methodology
+
+        In these tests, Inferno acts as an authorized resource server that queries the authorization server about an access 
+        token, rather than a client to a FHIR resource server as in the previous SMART App Launch tests.  
+
+        Ideally, Inferno should be registered with the authorization server as an authorized resource server
+        capable of accessing the token introspection endpoint through client credentials, per the SMART IG recommendations.  
+        However, the SMART IG only formally REQUIRES "some form of authorization" to access
+        the token introspection endpoint and does specifiy any one specific approach.  As such, the token introspection tests are 
+        broken up into two groups that can be run independently:
+
+        1. Token Introspection Request group - completes the introspection requests
+        2. Token Introspection Response group - validates the contents of the introspection responses
+
+        For ease of testing it is highly recommended to run the Token Introspection Request group after already completing the 
+        Standalone Launch tests, as Inferno will by default seek to introspect the access token received during those tests.
+        If this option is used, the Token Introspection Response test inputs will also auto-fill.
+
+        If needed, however, testers can:
+        
+        a. Provide a different active access token to be introspected in Token Introspection Request tests.  However, the
+        tester will need to manually input the access token response parameters for the Token Introspection Response tests. 
+        b. Skip the Token Introspection Request tests altogether and manually run the required access token AND introspection
+        requests out-of-band to complete the Token Introspection Response tests.  Given the extent of manual steps
+        and inputs required, we recommend this option only if it is not possible for the Inferno client to access the
+        token introspection endpoint.
       
-      group from: :token_introspection_test_group
-
-      run_as_group
+      )
+      
```

Comment:
Sorta annoyingly, we might want to do `input_instructions` at this level too, customized to what gets rolled up to here (which might get kinda long but I'm not sure how how to handle it?)

Also,  I think we might want to order these, just to make sure that the first 2 inputs are the URL and the bearer token, then expected fields from teh access token response, and then at the bottom we add in all that convoluted authorization stuff?  See https://inferno-framework.github.io/docs/writing-tests/test-inputs-outputs.html#ordering-inputs  (maybe same goes for the other tests?  auth stuff at the bottom as it just seems secondary to the purpose of the tests... this might just be a preferance thing)

Reply:
I ordered the inputs and added some instructions.  The good news is that IMO the inputs at this level are more self-explanatory, since running the introspection request out-of-band is not assumed.